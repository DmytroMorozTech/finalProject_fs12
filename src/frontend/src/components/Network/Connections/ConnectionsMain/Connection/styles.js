import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  connection: {
    display: 'flex',
    marginLeft: theme.spacing(5),
    marginRight: theme.spacing(5),
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2),
    justifyContent: 'space-between'
  },

  removed: {
    display: 'none'
  },

  flex: {
    display: 'flex'
  },

  avatar: {
    width: theme.avatar.large,
    borderRadius: '50%'
  },

  userInfo: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    marginLeft: theme.spacing(2)
  },

  link: {
    textDecoration: 'none',
    fontSize: theme.typography.sh3.fontSize,
    fontWeight: theme.typography.fontWeightBold,
    color: theme.palette.grey[700],
    '&:hover': {
      textDecoration: 'underline',
      textDecorationThickness: '2px',
      textDecorationColor: theme.palette.grey[700]
    }
  },

  buttons: {
    display: 'flex',
    alignItems: 'center'
  },

  button: {
    marginRight: theme.spacing(2)
  },

  line: {
    width: 'calc(100% - 100px)',
    height: '0.5px',
    border: '0',
    margin: '0',
    marginLeft: '100px',
    backgroundColor: theme.palette.grey[300]
  },

  linkButton: {
    textDecoration: 'none'
  }
}))
