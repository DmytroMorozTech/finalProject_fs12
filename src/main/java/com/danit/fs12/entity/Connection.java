package com.danit.fs12.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity(name = "Connection")
@Table(name = "connections")
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@Data
public class Connection extends AbstractEntity {

  @ManyToOne
  @JoinColumn
      (
          name = "user_id",
          nullable = false,
          referencedColumnName = "id",
          foreignKey = @ForeignKey(
              name = "user_connection_fk"
          )
      )
  private User user;

  @Column(
      name = "connected_user_id",
      updatable = false
  )
  private Long connectedUserId;
//   the user who was connected to the primary user (owner of connection)

  public Connection(User activeUser, Long userToBeConnectedId) {
    this.user = activeUser;
    this.connectedUserId = userToBeConnectedId;
  }
}
