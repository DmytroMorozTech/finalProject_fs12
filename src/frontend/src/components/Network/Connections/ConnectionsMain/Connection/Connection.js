import styles from './styles'
import avatar from '../../../../../temporaryImages/avatar.jpg'
import Avatar from '../../../../../shared/Avatar/Avatar'
import React, { useState } from 'react'
import Typography from '@material-ui/core/Typography'
import SharedButton from '../../../../../shared/Button/SharedButton'
import TreeDots from '../../../../../shared/ThreeDots/TreeDots'
import { Link } from 'react-router-dom'
import SimpleMenu from '../../../../../shared/PopupMenu/PopupMenu'
import ConnectionAddition from './ConnectionAddition/ConnectionAddition'

function Connection (props) {
  const classes = styles()

  const {
    avatarUrl = avatar,
    fullName = 'George Lupin',
    positionAndCompany = 'Fullstack Developer â€” GlobalLogic',
    dateCreated = '3 days ago'
  } = props

  const [removedConnection, setRemovedConnection] = useState(false)

  const handleRemoved = () => {
    setRemovedConnection(!removedConnection)
  }

  return (
    <div className={removedConnection ? classes.removed : ''}>
      <div>
        <div className={classes.connection}>
          <div className={classes.flex}>
            <Link to="/profiles/:id">
              <div className={classes.avatar}>
                <Avatar avatarUrl={avatarUrl}/>
              </div>
            </Link>
            <div className={classes.userInfo}>
              <Link to="/profiles/:id" className={classes.link}>
                {fullName}
              </Link>
              <Typography variant="body1" color="textSecondary">
                {positionAndCompany}
              </Typography>
              <Typography variant="h6" color="textSecondary">
                Connected {dateCreated}
              </Typography>
            </div>
          </div>
          <div className={classes.buttons}>
            <div className={classes.button}>
              <SharedButton title="Message" size="medium" variant="outlined"/>
            </div>
            <div>
              <SimpleMenu menuItem={
                <TreeDots/>
              } userData={<ConnectionAddition onClick={handleRemoved}/>}/>
            </div>
          </div>
        </div>
      </div>
      <hr className={classes.line}/>
    </div>
  )
}

export default Connection
