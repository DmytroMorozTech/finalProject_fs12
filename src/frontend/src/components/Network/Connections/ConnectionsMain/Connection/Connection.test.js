import React from 'react'
import Connection from './Connection'
import { shallow } from 'enzyme'
import * as redux from 'react-redux'

export const connection = {
  'id': 4,
  'createdDate': '2015-05-11T22:21:12.123123',
  'firstName': 'Everett',
  'lastName': 'Anderson',
  'fullName': 'Everett Anderson',
  'avatarPublicId': 'linkedin/avatars/i1bzhpqac3i1ld2urju8',
  'profileBgPublicId': '',
  'positionAndCompany': 'Project manager at HQSoftware',
  'email': 'test4@gmail.com',
  'isFollowedByActiveUser': true,
  'numberOfFollowers': 4
}

jest.mock('./styles', () => () => ({}))
jest.mock('react-router-dom', () => ({
  Link: 'a'
}))

const chats = [
  {
    'id': 1,
    'users': [
      {
        'id': 1,
        'createdDate': '2015-01-01T18:22:32.123123',
        'lastModifiedDate': '2021-10-12T00:00:00',
        'firstName': 'Richard',
        'lastName': 'West',
        'headline': 'Java Developer',
        'country': 'USA',
        'city': 'New York',
        'phoneNumber': '+380502926823',
        'email': 'test@gmail.com',
        'age': 20,
        'passwordHash': '$2a$10$VyCoVteS/iZ/5ZYTIGI0EOZEytTbZphdioSm0uXqNCQ29vK6giI0q',
        'avatarPublicId': 'linkedin/avatars/vbalexhudya0ano14pgs',
        'profileBgPublicId': 'linkedin/profile-bg/isgzneqfm5ro1bufycqc',
        'resetPasswordNumber': null,
        'provider': null,
        'fullName': 'Richard West',
        'numberOfFollowers': 2,
        'isFollowedByActiveUser': false,
        'positionAndCompany': 'Sales manager at Microsoft'
      },
      {
        'id': 2,
        'createdDate': '2015-03-03T22:22:32.123123',
        'lastModifiedDate': '2021-10-12T00:00:00',
        'firstName': 'Frank',
        'lastName': 'Jackson',
        'headline': 'Head of Sales Department',
        'country': 'Ukraine',
        'city': 'Kyiv',
        'phoneNumber': '+380674974924',
        'email': 'test2@gmail.com',
        'age': 32,
        'passwordHash': '$2a$10$ihFhDPp8TDs72rrtlD5tIewV7D6dvNGMXatbBz7e9zmrLDsHhai/O',
        'avatarPublicId': 'linkedin/avatars/kgpkn4a4fudfjpebyegx',
        'profileBgPublicId': 'linkedin/profile-bg/iltbrz2s8408pu0xrvsj',
        'resetPasswordNumber': null,
        'provider': null,
        'fullName': 'Frank Jackson',
        'numberOfFollowers': 3,
        'isFollowedByActiveUser': false,
        'positionAndCompany': 'Head of CIS region at Apple'
      }
    ]
  },
  {
    'id': 2,
    'users': [
      {
        'id': 1,
        'createdDate': '2015-01-01T18:22:32.123123',
        'lastModifiedDate': '2021-10-12T00:00:00',
        'firstName': 'Richard',
        'lastName': 'West',
        'headline': 'Java Developer',
        'country': 'USA',
        'city': 'New York',
        'phoneNumber': '+380502926823',
        'email': 'test@gmail.com',
        'age': 20,
        'passwordHash': '$2a$10$VyCoVteS/iZ/5ZYTIGI0EOZEytTbZphdioSm0uXqNCQ29vK6giI0q',
        'avatarPublicId': 'linkedin/avatars/vbalexhudya0ano14pgs',
        'profileBgPublicId': 'linkedin/profile-bg/isgzneqfm5ro1bufycqc',
        'resetPasswordNumber': null,
        'provider': null,
        'fullName': 'Richard West',
        'numberOfFollowers': 2,
        'isFollowedByActiveUser': false,
        'positionAndCompany': 'Sales manager at Microsoft'
      },
      {
        'id': 3,
        'createdDate': '2015-03-23T11:13:12.123123',
        'lastModifiedDate': '2021-10-12T00:00:00',
        'firstName': 'Laura',
        'lastName': 'Lee',
        'headline': 'Senior Java Developer',
        'country': 'Spain',
        'city': 'Barcelona',
        'phoneNumber': '+380677623175',
        'email': 'test3@gmail.com',
        'age': 62,
        'passwordHash': '$2a$10$2xFkfxj1Gx4PbJeqt12.V.dR2./r5fcojGGjtVPRZXa2JvWXuq0Ci',
        'avatarPublicId': 'linkedin/avatars/p1qwriz6hzjgwkaihwpa',
        'profileBgPublicId': 'linkedin/profile-bg/o0ngvhpnbu4bz4moqklt',
        'resetPasswordNumber': null,
        'provider': null,
        'fullName': 'Laura Lee',
        'numberOfFollowers': 3,
        'isFollowedByActiveUser': false,
        'positionAndCompany': 'Head of design Unit at Apple'
      }
    ]
  }
]

const setUp = (props) => shallow(<Connection {...props}/>)

describe('should render Connection component', () => {
  let spyOnUseSelector
  let spyOnUseDispatch
  let mockDispatch

  let component
  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(chats)
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp({connection})
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
