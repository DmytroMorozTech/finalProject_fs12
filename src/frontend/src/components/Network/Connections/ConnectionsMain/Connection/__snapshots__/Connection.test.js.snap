// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Connection component to match snapshot 1`] = `
<div>
  <div>
    <div>
      <div>
        <a
          to="/profiles/4"
        >
          <Image
            alt="user avatar"
            imageUrl="linkedin/avatars/i1bzhpqac3i1ld2urju8"
            type="largeAvatar"
          />
        </a>
        <div>
          <a
            to="/profiles/4"
          >
            Everett Anderson
          </a>
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            variant="body1"
          >
            Project manager at HQSoftware
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            variant="h6"
          >
            Connected 
            6 years
             ago
          </WithStyles(ForwardRef(Typography))>
        </div>
      </div>
      <div>
        <div>
          <a
            key="4"
            to="/chat/new/4"
          >
            <SharedButton
              size="medium"
              title="Message"
              variant="outlined"
            />
          </a>
        </div>
        <div>
          <SimpleMenu
            menuItem={<ThreeDots />}
            userData={
              <ConnectionAddition
                onClick={[Function]}
              />
            }
          />
        </div>
      </div>
    </div>
  </div>
  <hr />
</div>
`;
